extensions/azurecore/src/azurecore.d.ts

```diff
@@ -18,6 +18,49 @@ declare module 'azurecore' {
 		name = 'Microsoft.azurecore'
 	}
 
+	export interface AccountKey {
+		/**
+		 * Account Key - uniquely identifies an account
+		 */
+		key: string
+	}
+
+	export interface AccessToken extends AccountKey {
+		/**
+		 * Access Token
+		 */
+		token: string;
+	}
+
+	export interface RefreshToken extends AccountKey {
+		/**
+		 * Refresh Token
+		 */
+		token: string;
+
+		/**
+		 * Account Key
+		 */
+		key: string
+	}
+
+	export interface Token extends AccountKey {
+		/**
+		 * Access token
+		 */
+		token: string;
+
+		/**
+		 * Access token expiry timestamp
+		 */
+		expiresOn?: number;
+
+		/**
+		 * TokenType
+		 */
+		tokenType: string;
```

Comment:
As per documentation, the default type from AAD is 'Bearer', I'm not sure what other values could be possible. It doesn't make sense to restrict with enum, since we don't have a defined list.